### PR TITLE
feat: Use docker or podman to build and run the UI

### PR DESCRIPTION
**Pull request by rmarting**
This PR improves the `Makefile` to check which container tool is installed in the enviroment (only `docker` or `podman`) related with the issue #97.

The user will not have to set up anything before to build or to run the image locally, only a container tool is needed to be installed.

HTH

### CODE REVIEW

File: Makefile
Comment (HarikrishnanBalagopal) [inline Makefile:40]: We should use `command` instead of `which` similar to https://github.com/konveyor/move2kube/blob/main/scripts/install.sh#L42-L47
as recommended in https://mywiki.wooledge.org/BashFAQ/081

See also: https://unix.stackexchange.com/questions/85249/why-not-use-which-what-to-use-then

File: Makefile
Comment (HarikrishnanBalagopal) [inline Makefile:44]: we can change this to `CONTAINER_TOOL='docker'`

File: Makefile
Comment (HarikrishnanBalagopal) [inline Makefile:46]: `CONTAINER_TOOL='podman'`

File: Makefile
Comment (HarikrishnanBalagopal) [inline Makefile:106]: This `mkdir` should be on the podman side of the if else condition since `docker run` creates the directory if it doesn't exist.

Actually we can remove the if else block since `mkdir -p data` will succeed even if the `data` directory already exists.

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]
Some minor changes required. Other than that LGTM.

**Comment by ashokponkumar** [comment]
Thanks for the PR @rmarting ! 

Once this PR is merged, do you mind creating similar PRs for the following repos too : https://github.com/konveyor/move2kube/blob/main/Makefile
https://github.com/konveyor/move2kube-api/blob/main/Makefile
https://github.com/konveyor/move2kube-operator/blob/main/Makefile

File: Makefile
Comment (rmarting) [reply Makefile:40]: Indeed! My first implementation was using `command` however the GitHub Action fails because that command is not founded. So this is why I changed to use `switch`command.

You could see in this GitHub Action run: https://github.com/konveyor/move2kube-ui/runs/3741132624?check_suite_focus=true

Maybe I am doing sth wrong to use correctly the `command`.

**Comment by rmarting** [comment]
Thanks @ashokponkumar for your comments! I could do it in the other repos as soon this PR has been reviewed and accepted with the minor changes. Then I will be sure which changes I have to do in these repositories.

File: Makefile
Comment (HarikrishnanBalagopal) [reply Makefile:40]:
```
$ type command
command is a shell builtin
```
I think this is because Github actions uses a different shell instead of `bash`.
I will fix it in the workflow as shown here: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-4

Comment (rmarting) [reply Makefile]: Ok! Reverted my changes to use `command`. Could you check that now the PR is working successfully?

Comment (HarikrishnanBalagopal) [reply Makefile]: This should fix it https://github.com/konveyor/move2kube-ui/pull/99

Comment (HarikrishnanBalagopal) [reply Makefile]: @rmarting can you please change back to what you had initially with `command`?

**Review by rmarting — COMMENTED** [review]
Could you review it with my latest changes?

**Review by HarikrishnanBalagopal — APPROVED** [review]
LGTM

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]

File: Makefile
Comment (HarikrishnanBalagopal) [reply Makefile:49]: Also I have rebased the branch on top of main, we are using rebases instead of merges in this repo's commit history.

Comment (rmarting) [reply Makefile]: Done! WDTY?

Comment (HarikrishnanBalagopal) [reply Makefile]: Sorry, there's one more thing we missed. You need the `shell` prefix before `command`

Comment (HarikrishnanBalagopal) [reply Makefile]: That's the reason it was failing earlier.

Comment (HarikrishnanBalagopal) [reply Makefile]: @rmarting Thanks for making all the requested changes.
I am just debugging why `make` is saying it can't find `command`
Seems other people had the same issue https://github.com/cnabio/duffle/issues/502
Will merge after I fix this.

Comment (HarikrishnanBalagopal) [reply Makefile]: hmm https://stackoverflow.com/questions/12989869/calling-command-v-find-from-gnu-makefile/12991757

**Review by HarikrishnanBalagopal — APPROVED** [review]
LGTM again

File: Makefile
Comment (HarikrishnanBalagopal) [reply Makefile]: also https://stackoverflow.com/questions/5618615/check-if-a-program-exists-from-a-makefile

**Comment by rmarting** [comment]
Thank you so much for your support and review! Glad to contribute!

On the next days I will try to clone this changes in the other repos.

**Comment by ashokponkumar** [comment]
Thanks @rmarting !

**Pull request merged**
